connd/qconnectionagent.cpp

```diff
@@ -462,19 +481,25 @@ void QConnectionAgent::techChanged()
         tetheringWifiTech = 0;
         return;
     }
-    if (tetheringWifiTech) {
-        tetheringEnabled = tetheringWifiTech->tethering();
+    if (!netman->getTechnology("bluetooth")) {
+        tetheringBtTech = 0;
+        return;
     }
 
-    Q_FOREACH(NetworkTechnology *technology,netman->getTechnologies()) {
+    Q_FOREACH(NetworkTechnology *technology, netman->getTechnologies()) {
         if (!knownTechnologies.contains(technology->path())) {
             knownTechnologies << technology->path();
             if (technology->type() == "wifi") {
                 tetheringWifiTech = technology;
-                connect(tetheringWifiTech, &NetworkTechnology::poweredChanged, this, &QConnectionAgent::technologyPowerChanged);
-                connect(tetheringWifiTech, &NetworkTechnology::tetheringChanged,
-                        this, &QConnectionAgent::techTetheringChanged, Qt::UniqueConnection);
+            } else if (technology->type() == "bluetooth") {
+                tetheringBtTech = technology;
+            } else {
+                continue;
             }
+            connect(technology, &NetworkTechnology::poweredChanged, this, &QConnectionAgent::technologyPowerChanged);
+            connect(technology, &NetworkTechnology::tetheringChanged,
+                    this, &QConnectionAgent::techTetheringChanged, Qt::UniqueConnection);
```

Comment:
Do we really need to have member variables for both tetheringWifiTech and tetheringBtTech? Couldn't we just utilize technology->type() when need to compare, could we?

NB, member variables are missing m_-prefixes.

Reply:
I'll try to remove them and see how it goes. I've a feeling they're used to indicate whether the technology is available though.
None of the members of this class have an m prefix right now. It would be quite dramatic to change it.

Reply:
Ther are quite a few points where the status of each tech is checked (powered, connected, tethered etc). I could fetch them from netman each time, but they are also connected to slots and I'm not sure if we could rely on those connections persisting if they weren't members.